activestorage/test/service/mirror_service_test.rb

```diff
@@ -33,7 +33,7 @@ class ActiveStorage::Service::MirrorServiceTest < ActiveSupport::TestCase
 
   test "downloading from primary service" do
     data = "Something else entirely!"
-    key  = upload(data, to: SERVICE.primary)
+    key  = upload(data, to: @service)
```

Comment:
`SERVICE.primary` is correct here.

Reply:
Sorry.., I finally got all the intentions here!
I fixed the point above and also changed to pass io object instead of string since you cannot pass read io object directly to `SERVICE.primary`, which is a disk_service object.